Emscripten-build-instructions.md

```diff
@@ -0,0 +1,167 @@
+# Wasm Build Instructions
+
+## CppInterOp Wasm Build Instructions
+
+This document first starts with the instructions on how to build a wasm  
+build of CppInterOp. Before we start it should be noted that  
+unlike the non wasm version of CppInterOp we currently only  
+support the Clang-REPL backend using llvm>19 for osx and Linux.  
+We will first make folder to build our wasm build of CppInterOp.  
+This can be done by executing the following command
+
+```bash
+mkdir CppInterOp-wasm
+```
+
+Now move into this directory using the following command
+
+```bash
+cd ./CppInterOp-wasm
+```
+
+To create a wasm build of CppInterOp we make use of the emsdk  
+toolchain. This can be installed by executing (we only currently  
+support version 3.1.45)
+```bash
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install  3.1.45
+```
+
+Go back upto the top level build directory using
+
+```bash
+cd ../..
+```
+
+and activate the emsdk environment
+```bash
+./emsdk/emsdk activate 3.1.45
+source ./emsdk/emsdk_env.sh
+```bash
+
+Now clone the 19.x release of the LLVM project repository 
+and CppInterOp
+
+```bash
+git clone --depth=1 --branch release/19.x https://github.com/llvm/llvm-project.git
+git clone --depth=1 https://github.com/compiler-research/CppInterOp.git
+```
+
+Now move into the cloned llvm-project folder and apply  
+a required patch
+
+```bash
+cd ./llvm-project/
+git apply -v ../CppInterOp/patches/llvm/emscripten-clang${{ matrix.clang-runtime }}-*.patch
+```
+
+We are now in a position to build an emscripten build of llvm by executing the following
+```bash
+mkdir build
+cd build
+emcmake cmake -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_HOST_TRIPLE=wasm32-unknown-emscripten \
+                        -DLLVM_ENABLE_ASSERTIONS=ON                        \
+                        -DLLVM_TARGETS_TO_BUILD="WebAssembly" \
+                        -DLLVM_ENABLE_LIBEDIT=OFF \
+                        -DLLVM_ENABLE_PROJECTS="clang;lld" \
+                        -DLLVM_ENABLE_ZSTD=OFF \
+                        -DLLVM_ENABLE_LIBXML2=OFF \
+                        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+                        -DCLANG_ENABLE_ARCMT=OFF \
+                        -DCLANG_ENABLE_BOOTSTRAP=OFF \
+                        -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4" \
+                        ../llvm
+emmake make clang -j $(nproc --all)
+emmake make clang-repl -j $(nproc --all)
+emmake make lld -j $(nproc --all)
+```
```

Comment:
We have `cmake` in our `envoronment-wasm-build.yml` and I think we that should be setup and activate otherwise the emcmake cmake would fail !

Reply:
It will not fail. This is evident through the ci since it just builds our llvm with the non conda provided cmake.We are going to assume that for these build instructions

Reply:
We don't really need envoronment-wasm-build.yml since we can use the systems conda. See llvm build in the ci which does just that.

Reply:
Hmmm wait but we should atleast ask the user to get cmake from conda-forge using micromamba correct. 
I think you're right and the CI won't need it but if you do a micromamba install, I don't think you get cmake in your base env by default. 

Atleast I get a failure locally when I run a cmake command without having cmake installed

Reply:
I'll add it for now as I'm not too fussed about this.

Reply:
Just a note though that of course you'll get an error if you don't have cmake installed locally before executing the command.